src/components/TodoItem/Todo.item.tsx

```diff
@@ -0,0 +1,135 @@
+import classNames from 'classnames';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../TodoContext/TodoContext';
+import { Keyboard } from '../enum/Keyboard';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { title, id, completed } = todo;
+  const { todos, setTodos } = useContext(TodoContext);
+  const [isEditing, setIsEditing] = useState(false);
+  const [value, setValue] = useState(todo.title);
+
+  const replaceCompleted = (idTodo: number) => {
+    const newTodo = todos.map(item => {
+      if (item.id === idTodo) {
+        return {
+          ...item,
+          completed: !item.completed,
+        };
+      }
+
+      return { ...item };
+    });
+
+    setTodos(newTodo);
+  };
+
+  const titleFocus = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing) {
+      titleFocus.current?.focus();
+    }
+  }, [isEditing]);
+
+  const deleteTask = (idTodo: number) => {
+    setTodos(todos.filter(prevTodo => prevTodo.id !== idTodo));
+  };
+
+  const editTodos = () => {
+    setIsEditing(true);
+  };
+
+  const changeTudo = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  const handleBlurSeve = () => {
```

Comment:
when user starts editing and then doesn't submit this new value we shouldn't set this value. So don't save new value on blur. Only on submit